Quadriga/src/main/java/edu/asu/spring/quadriga/service/textfile/impl/TextFileService.java

```diff
@@ -0,0 +1,97 @@
+package edu.asu.spring.quadriga.service.textfile.impl;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+
+import edu.asu.spring.quadriga.dao.textfile.ITextFileDAO;
+import edu.asu.spring.quadriga.dao.workbench.IProjectDAO;
+import edu.asu.spring.quadriga.dao.workbench.IProjectWorkspaceDAO;
+import edu.asu.spring.quadriga.domain.workspace.ITextFile;
+import edu.asu.spring.quadriga.dto.TextFileDTO;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.textfile.ITextFileService;
+
+@PropertySource(value = "classpath:/user.properties")
+@Service
+public class TextFileService implements ITextFileService {
+
+    @Autowired
+    private TextFileDTO txtFileDTO;
+
+    @Autowired
+    private ITextFileDAO txtFileDAO;
+
+    @Autowired
+    private IProjectWorkspaceDAO projWSDAO;
+
+    @Autowired
+    private Environment env;
+
+    /* (non-Javadoc)
+     * @see edu.asu.spring.quadriga.service.textfile.ITextFileService#saveTextFile(edu.asu.spring.quadriga.domain.workspace.ITextFile)
+     * Service layer method to handle Text Management Operations
+     */
+    @Override
+    public boolean saveTextFile(ITextFile txtFile) throws QuadrigaStorageException, IOException {
+
+        UUID refId = UUID.randomUUID();
+        txtFile.setRefId(refId.toString());
+        saveTextFileLocal(txtFile);
+        saveTextFileDB(txtFile);
+
+        return true;
+    }
+
+    /**
+     * @param txtFile
+     *          TextFile object to be updated in the database
+     * @return
+     * @throws QuadrigaStorageException
+     */
+    private boolean saveTextFileDB(ITextFile txtFile) throws QuadrigaStorageException {
+        txtFileDTO.setFilename(txtFile.getFileName());
+        txtFileDTO.setProjectId(txtFile.getProjectId());
+        txtFileDTO.setRefId(txtFile.getRefId());
+        txtFileDTO.setWorkspaceId(txtFile.getWorkspaceId());
+
+        txtFileDAO.saveTextFileDTO(txtFileDTO);
+        return true;
+    }
+
+    /**
+     * @param txtFile
+     *          TextFile object to be updated in the FileSystem
+     * @return
+     * @throws IOException
+     */
+    private boolean saveTextFileLocal(ITextFile txtFile) throws IOException {
+
+        String saveDir = env.getProperty("textfile.location");
+        String filePath = saveDir + "/" + txtFile.getRefId();
+        File dirFile = new File(filePath);
+        dirFile.mkdir();
+
+        File saveTxtFile = new File(filePath + "/" + txtFile.getFileName() + ".txt");
```

Comment:
and when you submit a file from the controller you set the filename to be ending in 'txt' if it doesn't already.